src/Chirp.Razor/Shared/Storage/ChirpStorage.cs

```diff
@@ -40,29 +41,16 @@ public void StoreCheeps(List<Cheep> entities)
     {
     }
 
-    public List<Cheep> GetCheepsFromAuthor(int pageNumber, int amount, string author)
-    {
-        int startIndex = pageNumber * amount;
-        var cheeps = _db.Cheeps
-            .Skip(startIndex)
-            .Where(c => c.Author.Name == author)
-            .Include(c => c.Author)
-            .Take(amount)
-            .ToList();
-
-        return cheeps;
-    }
-
-    public IEnumerable<Cheep> GetCheepsPerPage(int pageNumber, int amount)
+    public List<Cheep> QueryCheeps(int pageNumber, int amount, string? author = null)
```

Comment:
Please change to IEnumerable.
This is done here and in the interface. You also need a tolist in cheepservice.